clients/drcachesim/simulator/cache_simulator.cpp

```diff
@@ -60,7 +60,15 @@ cache_simulator_create(const cache_simulator_knobs_t &knobs)
 analysis_tool_t *
 cache_simulator_create(const std::string &config_file)
 {
-    return new cache_simulator_t(config_file);
+    std::ifstream fin;
+    fin.open(config_file);
+    if (!fin.is_open()) {
+        ERRMSG("Failed to open the config file '%s'\n", config_file.c_str());
+        return nullptr;
+    }
+    analysis_tool_t *sim = new cache_simulator_t(&fin);
+    fin.close();
```

Comment:
Would it be better to use a unique_ptr so the cache_simulator itself can own the std::ifstream and close it on its own? I was a bit surprised to see the ifstream closed so soon after the constructor call.

Reply:
I like the idea but std::istream does not have a close() method, and std::ifstream is not simple to work with for our proprietary i/o library...per offline discussion going to live w/ the pointer.